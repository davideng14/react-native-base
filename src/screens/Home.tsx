import React from 'react';
import { View, Button } from 'react-native';

export default function Home(props: any) {
    return (
    <View>
        <Button title={"Home2"} onPress={() => {
            props.navigation.navigate('StoreScreen');
        }} />
    </View>
    )
} 