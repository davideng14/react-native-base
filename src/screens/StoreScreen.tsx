import React from 'react';
import { View, Button } from 'react-native';

export default function StoreScreen() {
    return (
    <View>
        
    </View>
    )
} 