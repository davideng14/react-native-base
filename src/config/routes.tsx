import Home from '../screens/Home';
import StoreScreen from '../screens/StoreScreen';

const routes = {
    home: { screen: Home }, 
    storeScreen: { screen: StoreScreen } 
}

export default routes;