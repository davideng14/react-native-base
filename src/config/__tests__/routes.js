import routes from '../routes';
import Home from '../../screens/Home';
import StoreScreen from '../../screens/StoreScreen';

it('has all needed routes', () => { 
    expect(routes).toMatchSnapshot(); 
});

it('has home screen', () => { 
    expect(routes.home.screen).toBe(Home); 
}); 

it('has storeScreen screen', () => { 
    expect(routes.storeScreen.screen).toBe(StoreScreen); 
});