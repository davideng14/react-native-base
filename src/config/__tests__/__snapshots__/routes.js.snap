// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has all needed routes 1`] = `
Object {
  "home": Object {
    "screen": [Function],
  },
  "storeScreen": Object {
    "screen": [Function],
  },
}
`;
